domain/application/state/unit.go

```diff
@@ -1689,3 +1755,18 @@ func (st *State) getModelConstraints(
 	}
 	return constraint, nil
 }
+
+func encodeResolveMode(mode sql.NullInt16) (string, error) {
+	if !mode.Valid {
+		return "none", nil
+	}
+
+	switch mode.Int16 {
+	case 0:
+		return "retry-hooks", nil
+	case 1:
+		return "no-hooks", nil
+	default:
+		return "", errors.Errorf("unknown resolve mode %d", mode.Int16).Add(coreerrors.NotSupported)
+	}
+}
```

Comment:
As we do with status, this should probably be imported from the `resolve` domain

Reply:
Maybe something for another day though, what do you think?

Reply:
Potentially, it's out of the scope of this PR.